src/main/java/org/radarcns/android/data/CacheStore.java

```diff
@@ -45,11 +46,26 @@ public static CacheStore getInstance() {
     }
 
     private final Map<String, CountedReference<DataCache>> caches;
+    private final SpecificData specificData;
     private SingleThreadExecutorFactory cacheExecutorFactory;
 
     private CacheStore() {
         caches = new HashMap<>();
         cacheExecutorFactory = null;
+        specificData = new SpecificData(CacheStore.class.getClassLoader()) {
+            @Override
+            protected boolean isFloat(Object object) {
+                return object instanceof Float
+                        && !((Float) object).isNaN()
+                        && !((Float) object).isInfinite();
+            }
```

Comment:
@sboettcher added a specific check for Float and Double, according to the test, this should make the method fail fast.

Reply:
Thanks! This way NaN records will just be omitted in the data without triggering the `didDisconnect()` of the Kafka connection, which is fine I think. Will try this out when I get a chance.

Reply:
Exactly. If you extend `AbstractDeviceManager`, it will log a message to console that an invalid record was attempted to be added + the topic name and record contents. If Firebase is used, it will add a non-fatal crash to Firebase Crashlytics.

If you want to handle an invalid record in another way, override the `send()` method of `AbstractDeviceManager` and catch an `IllegalArgumentException` when calling `DataCache.addMeasurement`.